src/Tools/ExternalAccess/FSharp/Internal/Editor/FSharpNavigationBarItemService.cs

```diff
@@ -53,24 +53,21 @@ private static ImmutableArray<NavigationBarItem> ConvertItems(IList<FSharpNaviga
             Document document, NavigationBarItem item, ITextView view, ITextVersion textVersion, CancellationToken cancellationToken)
         {
             // The logic here was ported from FSharp's implementation. The main reason was to avoid shimming INotificationService.
-            var navigationSpan = item.TryGetNavigationSpan(textVersion);
-            if (navigationSpan != null)
-            {
-                var span = navigationSpan.Value;
-                var workspace = document.Project.Solution.Workspace;
-                var navigationService = workspace.Services.GetRequiredService<IFSharpDocumentNavigationService>();
+            // Spans.First() is safe here as we filtered down to only items that have spans in ConvertItems.
+            var span = item.GetCurrentNavigationSpan(textVersion, item.Spans.First());
```

Comment:
This worries me a little just because it's a public method. There's no guarantee of call order right? Does `First()` provide a sufficient exception message to debug this if it happens?

Reply:
Same in `src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptNavigationBarItemService.cs`

Reply:
We effectively own the stack here.  this is basically an invariant of us already filtering down (Above) to only items that have at least one span.

note that the prior code was also presumed the same.